src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/api/aggregators/PersistableSlidingWindow.java

```diff
@@ -37,20 +38,25 @@
 public class PersistableSlidingWindow extends SlidingWindow<SlidingWindowData> {
   private static final Logger LOG = LogManager.getLogger(PersistableSlidingWindow.class);
   private static final ObjectMapper objectMapper = new ObjectMapper();
+  // The time to wait between writes, currently 5 minutes TODO (make this configurable)
+  private static final long WRITE_PERIOD_MS = 300000L;
 
   private Path pathToFile;
+  private boolean enablePersistence;
+  private long lastWriteTimeEpochMs;
 
   public PersistableSlidingWindow(int slidingWindowSize,
                                   TimeUnit timeUnit,
                                   Path filePath) {
     super(slidingWindowSize, timeUnit);
     this.pathToFile = filePath;
-    try {
-      if (Files.exists(pathToFile)) {
-        load(pathToFile);
-      }
-    } catch (IOException e) {
-      LOG.error("Couldn't load SlidingWindow data from {}", pathToFile, e);
+    if (this.pathToFile == null) {
+      this.enablePersistence = false;
```

Comment:
Minor: I feel like if we are giving this flag then this shouldn't really be a derived class, we should just add this ability to the base SlidingWindow class (which may be a good idea for a separate PR). Choosing a `PersistableSlidingWindow` instead of a `SlidingWindow` implies that you want persistence.

Reply:
You're right, I think the enablePersistence design is unnecessarily complicated to allow the caller to pass in a null path. 

The reason I built it this way was to avoid modifying all callers. We can and should change it in a future PR.